### ISSUE TITLE
Issue: Incorrect `requires_grad` in the last diagram

### ISSUE THREAD

**Issue by gggboy-9527** [pytorch/tutorials#3580]
In the tutorial [Understanding requires_grad, retain_grad, Leaf, and Non-leaf Tensors](https://docs.pytorch.org/tutorials/beginner/understanding_leaf_vs_nonleaf_tutorial.html), the **last diagram** (showing the computational graph) has an error.

It shows `z = x @ W + b` with `requires_grad = False`.  
However, since `W` and `b` both have `requires_grad = True`, the output `z` should **also** have `requires_grad = True`.

This is misleading for learners, as PyTorch automatically sets `requires_grad = True` for any tensor that depends on a tensor requiring gradients.

Please correct the diagram to reflect this.

Thank you!

<img width="851" height="436" alt="Image" src="https://github.com/user-attachments/assets/3fbcfeaf-c952-43aa-8a4b-049811936baa" />

**Comment by sekyondaMeta** [comment]
For Docathon 2026 if working on thiis issue, can you copy both the diagrams in this tutorial and recreate them as mermaid diagram.
